homeassistant/components/fitbit/sensor.py

```diff
@@ -357,9 +370,24 @@ def __init__(
         self.config_path = config_path
         self.is_metric = is_metric
         self.clock_format = clock_format
+        self.include_device = include_device
         self.extra = extra
+
+        if include_device:
+            self._attr_has_entity_name = True
+            self._attr_device_info = DeviceInfo(
```

Comment:
Only integrations supporting config entries and config flow can use device info.

https://developers.home-assistant.io/docs/device_registry_index#automatic-registration-through-an-entity

Reply:
Oh, I'd forgotten about that.
I merged #79587 so now at least users can update the entity names manually.

I guess @ndbroadbent would need to first add config flow before then adding device info.

Reply:
Ahhh I see, that makes sense! Would you be able to point me to any existing config flows that already implement an Oauth authentication? I'd be happy to implement that if I can find some conventions or existing code to copy

Reply:
You can make a search for config_entry_oauth2_flow.
Please note that config_flow implementations needs to have 100% test coverage.

Reply:
I've got a branch for config flow for fitbit here:
https://github.com/MartinHjelmare/home-assistant/tree/add-fitbit-oauth-flow

I haven't submitted it yet since there's a connection request error that I don't understand.